core/src/main/scala/org/apache/spark/rpc/RpcEndpointRef.scala

```diff
@@ -92,4 +92,18 @@ private[spark] abstract class RpcEndpointRef(conf: SparkConf)
     timeout.awaitResult(future)
   }
 
+  def askSyncWithInterruptCheck[T: ClassTag](
```

Comment:
Maybe just put this method to BarrierTaskContext only? I think it's rarely used for other cases, it's less necessary to put it in RpcEndpointRef.

Reply:
@xuanyuanking  I found an issue. If we check outside the RPC request future, we have no way to close RPC connection when it is killed. So see my PR latest update.